### ISSUE TITLE
Problem with fetching not a json features

### ISSUE THREAD

**Issue by melniqw** [growthbook/growthbook-kotlin#161]
![Clipboard - 17 декабря 2024 г , 18_51](https://github.com/user-attachments/assets/51db112e-fe57-42fb-ad0d-c4d07f36667a)

We have faced a problem in the lib code caused by the marked string. If the features receive not a json string (for some reason) we fallback to crash right after initialization.

**Comment by Bohdan-Kim** [comment]
Not a json features but what the features?
My feature type was String and it was fetched successfully

**Comment by melniqw** [comment]
Yeah I just figured out one more thing, it gets crashed even if we got 500.
Because onSuccess is still being called after error, so check does not stop the block actually. 
<img width="720" alt="Снимок экрана 2024-12-17 в 16 05 12" src="https://github.com/user-attachments/assets/acec3e10-1eda-46da-bbc0-c9bad3f8fd6f" />
And the features request was a second step of getting malformed answer.

**Comment by Bohdan-Kim** [comment]
It should be checked.
Do you have ability to use NetworkDispatcher based on Ktor instead of OkHttp?

**Comment by melniqw** [comment]
I came to a solution to write my custom NetworkDispatcher with the fixes in **consumeGETRequest** and **consumePOSTRequest** methods:
<img width="538" alt="Снимок экрана 2024-12-18 в 02 31 19" src="https://github.com/user-attachments/assets/6f4173e7-644e-4a17-9706-4450d4c6b9b7" />

**Comment by Bohdan-Kim** [comment]
Great!

**Comment by Bohdan-Kim** [comment]
> Yeah I just figured out one more thing, it gets crashed even if we got 500. Because onSuccess is still being called after error, so check does not stop the block actually. <img alt="Снимок экрана 2024-12-17 в 16 05 12" width="720" src="https://private-user-images.githubusercontent.com/3869148/396507939-acec3e10-1eda-46da-bbc0-c9bad3f8fd6f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU2Mzg2MDMsIm5iZiI6MTczNTYzODMwMywicGF0aCI6Ii8zODY5MTQ4LzM5NjUwNzkzOS1hY2VjM2UxMC0xZWRhLTQ2ZGEtYmJjMC1jOWJhZDNmOGZkNmYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMzFUMDk0NTAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDNmNjc3ZWFhY2UwYTMwZGFhMzc1YWI4OTZkYTVmOGFiMGE2OWQyN2RhN2ViYzhkYWJiZDFlYWNjNzA5ZWFlNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.eXRadsedlralNycLIqPCYKXvPESn9_lAMVGluex_GQo"> And the features request was a second step of getting malformed answer.
> 
> About the features response: still there is a possibility to get something unexpected even with 200 code, so handling the possibility of bad json when decoding is still a good practice imo

It is screenshot from OkHttpDispatcher v1.0.0; v1.0.2 is available on MavenCentral

**Comment by vazarkevych** [comment]
Hi, @melniqw. Have you tried increasing the version? If that resolves the issue, please close the issue.
